snuba/query/processors/events_column_processor.py

```diff
@@ -1,12 +1,6 @@
-from snuba.query.conditions import (
-    binary_condition,
-    BooleanFunctions,
-    ConditionFunctions,
-)
-from snuba.query.dsl import literals_tuple
+from snuba.clickhouse.processors import QueryProcessor
+from snuba.clickhouse.query import Query
```

Comment:
I don't think `snuba.query.processors` is the right place for these physical query processors at this point.

I think this should actually move into the file containing the events dataset or storage:

1. it's very coupled with that physical schema
2. that schema is also coupled with ClickHouse
3. this seems to have very little reuse value (so it doesn't belong in `snuba.clickhouse`)

It probably makes sense to do this as a follow up to minimize this specific change, though.

More broadly speaking, I think in general we should start more seriously splitting the code that is reusable (dataset classes, processor interfaces, etc) from the code that isn't (dataset instances, processors subclasses, etc) since those boundaries are becoming more clearly defined. (There is probably a better way to describe this logical separation but I can't find the right words this morning.) Personally, I find it a bit disorienting when reviewing that the dataset instances (which are almost practically configuration) are defined next to the dataset "framework" itself at a sibling level for example.

Reply:
Yeah I agree we need to take a stab to a consistent reorganization of the code base. 
I could start by moving this file to the datasets/storages directory. That's where the events storage is.